torch/distributed/_fsdp/fully_sharded_data_parallel.py

```diff
@@ -651,6 +730,13 @@ def _wait_for_post_backward(self) -> None:
             self._assert_state(TrainingState_.BACKWARD_PRE)
 
         torch.cuda.current_stream().wait_stream(self._streams["post_backward"])
+        if self.cpu_offload.offload_params:
+            # We need to wait for the non-blocking GPU ->
+            # CPU grad transfers to finish. TODO investigate why this is needed
```

Comment:
I think this is because, when the grad is on CPU, CPU comp won't wait for any CUDA stream (including default stream), unless we explicitly block CPU thread using `torch.cuda.current_stream().synchronize()`.

Reply:
Ah, that makes sense. I'll update the comment here to avoid confusion later.